pkgs/by-name/gi/git-warp-time/package.nix

```diff
@@ -0,0 +1,78 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+
+  # nativeBuildInputs
+  zstd,
+  pkg-config,
+  jq,
+  cargo,
+  rustc,
+  rustPlatform,
+
+  # buildInputs
+  libgit2,
+  typos,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "git-warp-time";
+  version = "0.8.4";
+
+  src = fetchurl {
+    url = "https://github.com/alerque/git-warp-time/releases/download/v${finalAttrs.version}/git-warp-time-${finalAttrs.version}.tar.zst";
+    sha256 = "sha256-Xh30nA77cJ7+UfKlIslnyD+93AtnQ+8P3sCFsG0DAUk=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
+    nativeBuildInputs = [ zstd ];
+    dontConfigure = true;
+    hash = "sha256-ozy8Mfl5fTJL2Sr22tCSnK30SOKaC9cL+g4lX6ivi9Q=";
+  };
+
+  nativeBuildInputs = [
+    zstd
+    pkg-config
+    jq
+    cargo
+    rustc
+    rustPlatform.cargoSetupHook
+  ];
+
+  buildInputs = [
+    libgit2
+    typos
+  ];
+
+  env = {
+    LIBGIT2_NO_VENDOR = "1";
+  };
+
+  outputs = [
+    "out"
+    "doc"
+    "man"
+    "dev"
+  ];
+
+  enableParallelBuilding = true;
+
+  meta = {
+    description = "Utility to reset filesystem timestamps based on Git history";
+    longDescription = ''
+      A CLI utility that resets the timestamps of files in a Git repository
+      working directory to the exact timestamp of the last commit which
+      modified each file.
+    '';
```

Comment:
Very cool! I'm surprised this is the 1st time I'm hearing of this project.

Reply:
Glad to hear you might get some use out of it. It certainly isn't useful for all workflows and will actually jinx up some build systems, but when you have a use case (e.g. a timestamped based build system with a shared cache or permanent output location and long build times) it's indispensable.

It's been around for 4 years or so with a working flake based partially on your help with other projects. As far as I know there is only one similar tool out there, a shell script with somewhat dubious dependencies and very long run times.